### ISSUE TITLE
This is the official repository, most updated??

### ISSUE THREAD

**Issue by francot514** [piorrro33/psxjin#67]
Hey just found this interesting emulator with lua scripting and want to be sure if this is official one and the mos updated.

**Comment by piorrro33** [comment]
Hi, yes this is. We haven't added so much but yes, feel free to grab the latest version from the releases page. :)

**Comment by francot514** [comment]
Was asking because seems save states failed to work..

**Comment by piorrro33** [comment]
Hmm this is weird. Do you have any error message or something? I'll have a look at it too then I'll report back.

**Comment by piorrro33** [comment]
Okay so I just looked into it and it seems like the savestate feature is a bit broken (like, if you make a savestate, it'll work halfly).
I'll have a look at the older versions to see what broke it (if it ever worked fine!).

**Comment by francot514** [comment]
Want me to edit issue name to clarify about the bug? Also this can be fixed?

**Comment by piorrro33** [comment]
Yeah, if you want. But I just looked at Google Code and it seems like the repo is gone, so we'd have to build the app manually to see if it worked. Also as of what I saw, savestate was supposed to be a new feature so it does not really surprise me (look at the last commits from the original team).

**Comment by francot514** [comment]
I was looking for that, but seems there is no info about changes, seems uncomplete feature

**Comment by piorrro33** [comment]
Yep, sadly. Also this project is really ded, you might want to check out ePSXe, it's much better.

**Comment by francot514** [comment]
Epxse?? that is closed source and no savestate feature.

**Comment by vgturtle127** [comment]
@francot514 Save states were mostly broken in the original version as well, at least on my rig. 😨 

@piorrro33 I think this is another issue that the original project had, which I think we should switch to another codebase. libretro-beetle, maybe. Slap a GUI on it and call it good?

**Comment by piorrro33** [comment]
@vgturtle127 Oh hello there, long time no see.
Yes, we can do that.

I agree, it was an interesting project. Also no it's not so hard to get older working versions. Simply revert to the commit you want on your local repo. Of course you'd probably have to fix stuff that was using the SVN version parser (pretty easy to fix) then you could build the older version.

**Comment by vgturtle127** [comment]
> SVN version parser

The horrible nightmares of trying to improve this code base are coming back. 😆 

@piorrro33 Me and you should talk soon, but that's what I was thinking. Not sure what language it is in, and what sort of stuff we would have to do with it, but I want to research it and see if it is a good option.

**Comment by piorrro33** [comment]
@vgturtle127 Honestly it worked pretty well. The Git parser I used when we moved worked just fine too, haha.
Yeah sure, why not. Just contact me wherever you want (Piorrro33 anywhere).

**Comment by francot514** [comment]
So no workaround or future improvements for this feature?

**Comment by piorrro33** [comment]
@francot514 Not in the close the future at least. You may as well code it yourself, but huh, that's going to be a challenge. :/

**Comment by vgturtle127** [comment]
@francot514 Actually, I may work on psxjin as a side project, so I wouldn't say never or anything.

But to be honest this emulator isn't accurate enough to warrant using save states all the time. A better option would be just to use the built-in saves in the game. This could change slightly in the future, but we may primarily switch to a different code base, which may not have the issue anyway.

If you use save states in this, they have a tendency to get corrupted, especially on some longer games, like RPG's (the one place you wouldn't want to lose your save state).

**Comment by francot514** [comment]
Ok, so ill wait for later, if you decide to do something., is not like im able to implement that without enought knowledge.

**Comment by vgturtle127** [comment]
@francot514 Have you tried other emulators? ePSXe, PCSX-Reloaded, Mednafen, etc.?
